app/Repo/MenuRepository.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Repo;
+
+use App\Models\Menu;
+
+class MenuRepository extends BaseRepository implements MenuRepositoryInterface
+{
+    /**
+     * @var Menu
+     */
+    protected $model;
+
+    /**
+     * PostRepository constructor.
+     *
+     * @param Post $model
+     */
+    public function __construct(Menu $model)
+    {
+        parent::__construct($model);
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getList()
+    {
+        $listMenu = Menu::with('parent:id,name')->orderBy('lft')->get();
```

Comment:
biến `$model` trên kia dùng làm gì mà không query vào đây em?

Reply:
biến $model ở trên em chỉ dùng để khởi tạo lớp cha thôi, biến $model sẽ được Laravel resolve ra 1 đối tượng Menu rỗng và em sẽ truyền nó vào hàm khởi tạo của thằng cha là BaseRepository, ở trong đó có các hàm chung mà Model nào cx sử dụng như save(), find() ... Thay vì dùng Menu::find() hoặc Menu::all() em có thể dùng $this->model->find() hoặc $this->model->all()
2 cách này là tương đương nhau ạ

Reply:
thế sao không dùng `$this->model->with('parent:id,name')->orderBy('lft')->get();`.
em đã mất công inject model vào đây mà lại dùng query builder kiểu `Menu::with()...` thì cần gì dùng đến pattern nữa

Reply:
oki anh, em hiểu rồi ạ, đoạn này em bị ngáo tí :v